pkg/kn/config/config.go

```diff
@@ -119,17 +126,41 @@ func BootstrapConfig() error {
 		return err
 	}
 
+	if globalConfig.skipInitCfg && globalConfig.reInitCfg {
+		return fmt.Errorf("only one of --no-init and --init may be specified")
+	}
+
+	if globalConfig.skipInitCfg {
+		return nil
+	}
+
 	// Check if configfile exists. If not, just return
 	configFile := GlobalConfig.ConfigFile()
 	_, err = os.Lstat(configFile)
-	if err != nil {
-		if os.IsNotExist(err) {
-			// No config file to read
-			return nil
+
+	if err != nil && os.IsNotExist(err) {
+		cfgDir := filepath.Dir(configFile)
+		if err := os.MkdirAll(cfgDir, 0775); err != nil && !os.IsExist(err) {
+			return fmt.Errorf("Failed creating directory path %s: %w", cfgDir, err)
+		}
+		if err := viper.SafeWriteConfigAs(configFile); err != nil {
+			return fmt.Errorf("Failed saving initial file %s: %w", configFile, err)
 		}
+		return nil
```

Comment:
We probably shouldn't exit the functions here or at least have desired defaults in `ConfigFile`, since the `ConfigFile` is initialized now.

Let's have a case:
1. No config file
2. First `kn plugin list` call will produce the config file and desired list
3. Second `kn plugin list` uses incorrect defaults therefore list is empty

However, in case of no. 2 clearly the new file wasn't used, as L164-165 are skipped due to the `exit`. 

```
➜  client git:(closes_1232) ls -l ~/.config/kn/
total 16
-rw-r--r--  1 dsimansk dsimansk   46 Oct  5 11:07 config.yaml.bak
-rw-r--r--  1 dsimansk dsimansk  104 Aug  4 19:37 config.yaml.bkup.2021-10-05T10:57:48+02:00
-rw-r--r--  1 dsimansk dsimansk   46 Oct  5 11:07 config.yaml.bkup.2021-10-05T11:07:47+02:00
drwxr-xr-x. 2 dsimansk dsimansk 4096 Feb 24  2021 plugins
➜  client git:(closes_1232) kn plugin list       
- kn-func : /home/linuxbrew/.linuxbrew/bin/kn-func
- kn-hello.sh : /home/dsimansk/.config/kn/plugins/kn-hello.sh
- kn-nox : /home/dsimansk/.config/kn/plugins/kn-nox
- kn-source-kafka : /home/dsimansk/.config/kn/plugins/kn-source-kafka

WARNING:
  - /home/dsimansk/.config/kn/plugins/kn-nox is not executable by current user
➜  client git:(closes_1232) kn plugin list
No plugins found.
➜  client git:(closes_1232) 
```